tests/Test/Compiler.hs

```diff
@@ -53,21 +52,21 @@ testIf handleResult filePaths =
   where
     setupTest filePath =
       do  source <- readFile filePath
-          let result = Compiler.compile "elm-lang" "core" source Map.empty
+          let result = Compiler.compile "elm-lang" "core" True source Map.empty
           return (testCase filePath (handleResult result))
 
 
 -- CHECK RESULTS
 
-isSuccess :: Either String a -> Assertion
-isSuccess result =
+isSuccess :: ([Compiler.Warning], Either [Compiler.Error] a) -> Assertion
+isSuccess (_, result) =
     case result of
-      Right _ -> assertBool "" True
-      Left msg -> assertFailure msg
+      Right _     -> assertBool "" True
+      Left errors -> assertFailure $ (unlines $ map (Compiler.errorToString "" "") errors)
```

Comment:
The function ([here](https://github.com/elm-lang/elm-compiler/blob/master/src/Elm/Compiler.hs#L90)) takes in the file path as the first argument and the source code for the second argument. It will fail if you do not provide the source code.

It looks like both of these things are available, but this code needs to be slightly reworked for those values to flow to this point.

You can also use `concatMap` instead of `unlines . map` because it adds the appropriate amount of spaces already.